lib/srv/regular/sshserver_test.go

```diff
@@ -446,14 +446,14 @@ func (s *SrvSuite) TestAgentForward(c *C) {
 	c.Assert(err, IsNil)
 	// clt must be nullified to prevent double-close during test cleanup
 	s.clt = nil
-	for i := 0; i < 4; i++ {
+	for i := 0; i < 10; i++ {
```

Comment:
This is definitely just a band-aid rather than a proper fix. I don't have enough insight into the internals yet.
Currently I imagine I could  add to the list of closers that are maintained by the `ConnectionContext`:
```go
agentDoneChan := make(chan struct{})
c.srv.agentClosers = append(c.srv.agentClosers, closerFunc(func() error {
	close(agentDoneChan)
}))
// agentClosers are additional closers ConnectionContext will close.

// make sure the socket is gone after we closed the connection.
err = s.clt.Close()
c.Assert(err, IsNil)
// clt must be nullified to prevent double-close during test cleanup
s.clt = nil

select {
case <-agentDoneChan:
case <-time.After(1 * time.Second):
	c.Fatalf("expected socket to be closed, still could dial after 1 s")
}
```
even though it'd rely on a timeout it avoids polling. It does rely on the implementation detail about the directory closer that is supposed to remove the socket and should run before the closer from the test.